lib/env/database.js

```diff
@@ -1,18 +1,35 @@
 const { MongoClient } = require("mongodb");
+const mongodb_url_parser = require('mongodb/lib/url_parser.js');
 const _ = require("lodash");
 const configFile = require("./configFile");
 
 module.exports = {
   async connect() {
     const config = await configFile.read();
     const url = _.get(config, "mongodb.url");
-    const databaseName = _.get(config, "mongodb.databaseName");
+    let databaseName = _.get(config, "mongodb.databaseName");
     const options = _.get(config, "mongodb.options");
 
     if (!url) {
       throw new Error("No `url` defined in config file!");
     }
 
+    if (!databaseName) {
+      databaseName = await new Promise(function(resolve, reject) {
```

Comment:
This part of the code can be changed like so:
```typescript
const { promisify } = require('util')
const result = await promisify(mongodb_url_parser)(url)
result.dbName
```
Also you can remove `.js` extension when loading files from the system with `require()`

```
const mongodb_url_parser = require('mongodb/lib/url_parser');
```

Reply:
I prefer to be precise there (.js extension).

Reply:
Thanks for the feedback, I will test this if it behaves in all cases as expected. I prefer to have the control (about resolve and reject). Also `result` may be different in this case and `dbName` not set in all cases, especially when an invalid URL is provided.

Reply:
Especially https://github.com/mongodb/node-mongodb-native/blob/master/lib/url_parser.js#L134 is only setting these two parameters in the callback. Nothing that directly resolved or rejects a promise and `dbName` might not be set then.

Did you tests your code or how did you test it to ensure that it works as expected?

Reply:
Especially https://github.com/mongodb/node-mongodb-native/blob/master/lib/url_parser.js#L134 is only setting these two parameters in the callback. Nothing that directly resolved or rejects a promise and `dbName` might not be set then.

Did you test your code or how did you test it to ensure that it works as expected?